core/network/address.go

```diff
@@ -243,28 +243,23 @@ func (a MachineAddress) sortOrder() int {
 	return order
 }
 
-// NewMachineAddress creates a new MachineAddress, deriving its type from the
-// value and using ScopeUnknown as scope. It is a shortcut to calling
-// NewScopedMachineAddress(value, ScopeUnknown).
-func NewMachineAddress(value string) MachineAddress {
-	return NewScopedMachineAddress(value, ScopeUnknown)
-}
-
-// NewScopedMachineAddress creates a new MachineAddress, deriving its type from
-// the value.
-// If the specified scope is ScopeUnknown, then NewScopedSpaceAddress will attempt
-// to derive the scope based on reserved IP address ranges.
-// Because passing ScopeUnknown is fairly common,
-// NewMachineAddress() above does exactly that.
-func NewScopedMachineAddress(value string, scope Scope) MachineAddress {
+// NewMachineAddress creates a new MachineAddress,
+// applying any supplied options to the result.
+func NewMachineAddress(value string, options ...func(AddressMutator)) MachineAddress {
```

Comment:
... and the constructors are changed in this file.

Everything else is dross to accommodate the new form of address construction.